### PR TITLE
Tabs Component

### LINKED ISSUE #2304: Add horizontal tabs clarity component
**Describe the problem/challenge you have**
We want to present horizontal tabs to the user. Worth noting that the specific implementation we have in mind involves putting the tabs inside of an accordion component

**Describe the solution you'd like**
Implement horizontal tabs from the clarity tabs component. 

**Anything else you would like to add:**
Clarity Docs: https://clarity.design/angular-components/tab/#horizontal

### PR DESCRIPTION
**Pull request by ftovaro**
Signed-off-by: ftovaro <f.tovar12@gmail.com>

**What this PR does / why we need it**:
Creates tabs component

image

**Which issue(s) this PR fixes**
- Fixes #2304 

**Special notes for your reviewer**:

**Release note**:
```
release-note
```

### CODE REVIEW

**Review by mklanjsek — CHANGES_REQUESTED** [review]
Looking good, just a few small changes in presentation layer.

File: web/src/stories/tabs.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+<Canvas withToolbar>
+   <Story name="Vertical Tabs component"
+          parameters={{ docs: tabsVerticalDocs }}
+          args= {{ view: tabsVerticalView }}>
+     { TabsStoryTemplate.bind({}) }
+   </Story>
+</Canvas>
+
+<h2>Props</h2>
+<ArgsTable story = "Tabs component" />
```
[the remark this answers is not in the document]
Comment (ftovaro) [reply web/src/stories/tabs.stories.mdx]: Good catch, thank you

File: web/src/app/modules/shared/components/presentation/tabs-view/tabs-view.component.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  display: flex;
+  justify-content: space-between;
+  flex-wrap: wrap;
+  max-width: 10rem;
+  margin-left: 0.8rem;
+
+  .item {
+    margin: 0 5px;
+  }
+}
```
[the remark this answers is not in the document]
Comment (ftovaro) [reply web/src/app/modules/shared/components/presentation/tabs-view/tabs-view.component.scss]: Done

File: web/src/app/modules/shared/components/presentation/tabs-view/tabs-view.component.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  tabs: View[] = [];
+  orientation: string;
+
+  constructor() {
+    super();
+  }
+
+  update() {
+    this.tabs = this.v.config.tabs;
+    if (this.v.config.orientation === 'vertical') {
```
[the remark this answers is not in the document]
Comment (ftovaro) [reply web/src/app/modules/shared/components/presentation/tabs-view/tabs-view.component.ts]: This is embarrassing, I was trying to do something different and forgot this little piece of code, thank you

**Comment by mklanjsek** [comment]
@ftovaro sorry for late addition, but that button can't touch the tabs in horizontal layout - there needs to be some spacing around `app-view-container`.  The easiest way to deal with that would be to wrap it inside `div` and conditionally set the top margin to 8 px for horizontal layout.

**Comment by ftovaro** [comment]
@mklanjsek Thanks, I was dealing with that and with you comment I solved easily.

image

**Review by mklanjsek — APPROVED** [review]
Looking good!

File: pkg/view/component/tabs_view.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	}
+
+	for _, tab := range x.Tabs {
+		c, err := tab.Contents.ToComponent()
+		if err != nil {
+			return err
+		}
+		fl, ok := c.(*FlexLayout)
+		if !ok {
+			return fmt.Errorf("item was not a tab")
```
Comment (GuessWhoSamFoo) [inline pkg/view/component/tabs_view.go:82]: Error should be flexlayout

**Pull request merged**